pkgs/development/libraries/modern-cpp-kafka/default.nix

```diff
@@ -0,0 +1,52 @@
+{ rdkafka
+, stdenv
+, lib
+, fetchFromGitHub
+, fetchpatch
+, cmake
+, boost
+, gtest
+, rapidjson
+}:
+
+stdenv.mkDerivation rec {
+  pname = "modern-cpp-kafka";
+  version = "2023.03.07";
+
+  src = fetchFromGitHub {
+    repo = "modern-cpp-kafka";
+    owner = "morganstanley";
+    rev = "v${version}";
+    hash = "sha256-7hkwM1YbveQpDRqwMZ3MXM88LTwlAT7uB8NL0t409To=";
+  };
+
+  patches = [
+    (fetchpatch {
+      name = "fix-avoid-overwriting-library-paths.patch";
+      url = "https://patch-diff.githubusercontent.com/raw/morganstanley/modern-cpp-kafka/pull/221.patch";
```

Comment:
Can you use the url `https://github.com/morganstanley/modern-cpp-kafka/pull/221.patch`, which redirects to this url? That way it will still work if GitHub decides to redirect to a different one.

Reply:
good idea; done